src/SearchQueryBuilder.js

```diff
@@ -0,0 +1,253 @@
+                    : builder.orWhere(property, content);
+            case 'neq':
+            case 'gt':
+            case 'lt':
+            case 'gte':
+            case 'lte':
+            case 'like':
+            case 'ilike':
+            case 'nlike':
+            case 'nilike':
+                const mappedOperator = operatorMap[operator];
+                return isAnd
+                    ? builder.where(property, mappedOperator, content)
+                    : builder.orWhere(property, mappedOperator, content);
+            case 'between':
+                return isAnd
+                    ? builder.whereBetween(property, content)
+                    : builder.orWhereBetween(property, content);
+            case 'inq':
+                return isAnd
+                    ? builder.whereIn(property, content)
+                    : builder.orWhereIn(property, content);
+            case 'nin':
+                return isAnd
+                    ? builder.whereNotIn(property, content)
+                    : builder.orWhereNotIn(property, content);
+            default:
+            }
+        }
+        // @todo: throw an error? silently ignore it?
+    }
+
+    applyJoins(rootModel, relationsToJoin, builder) {
+
+        return relationsToJoin.reduce((aliasedModels, relation) => {
+
+            const throughModel = relation.modelThrough;
+            const toAlias = this.createAlias(rootModel.getName(), relation);
+            const modelToQuery = ModelWrapper.fromModel(relation.modelTo, toAlias);
+
+            aliasedModels[relation.name] = modelToQuery;
+
+            // belongs to and has many without through model work exactly the same
+            if (!throughModel) {
+            // pure has many or belongTo
+                this.joinEntities(relation, rootModel, modelToQuery, builder);
+            } else {
+            // @todo: move this into a method to reduce complexity
+                const throughAlias = this.createAlias(rootModel.getName(), relation, true);
+                const modelThroughQuery = ModelWrapper.fromModel(throughModel, throughAlias);
+
+                // two joins required: one on the through model, one on the to model
+                this.joinEntities(relation, rootModel, modelThroughQuery, builder);
+
+                const [targetModelId] = modelToQuery.getIdProperties({ ignoreAlias: true });
+                const relationTargetProperty = modelToQuery.getPropertyQueriedThrough(relation);
+
+                const virtualRelation = {
+                    keyFrom: relation.keyThrough,
+                    keyTo: relationTargetProperty || targetModelId,
+                };
+
+                this.joinEntities(virtualRelation, modelThroughQuery, modelToQuery, builder);
+            }
+            return aliasedModels;
+        }, {});
+    }
+
+    createRootQuery(builder, rootModel, filter = {}) {
+
+        const [id] = rootModel.getIdProperties();
```

Comment:
as i understand it the you are making some assumptions when getting the id properties? make sure to document this in the readme or somewhere where it's highly visible.